state/storage.go

```diff
@@ -295,6 +298,34 @@ func (sb *storageBackend) AllStorageInstances() ([]StorageInstance, error) {
 	return out, nil
 }
 
+// DeleteStoragePool removes a pool only if its not currently in use
+func (sb *storageBackend) DeleteStoragePool(poolName string) error {
+	storageCollection, closer := sb.mb.db().GetCollection(storageInstancesC)
+	defer closer()
+
+	var inUse bool
+	if sb.modelType == ModelTypeCAAS && poolName == caas.OperatorStoragePoolName {
```

Comment:
Add TODO to say that we need to change the data model to support doing this as an assert